### ISSUE TITLE
Implement ssl_verify and SOCKS network options

### ISSUE THREAD

**Issue by wolfv** [QuantStack/mamba#151]

**Comment by wolfv** [comment]
SSL Verify == false is implemented now, but the certificate files options still need work.

**Comment by wolfv** [comment]
Ok, I just figured out that I can probably use Charles Proxy to debug this :)

https://www.charlesproxy.com/documentation/using-charles/ssl-certificates/

**Comment by wolfv** [comment]
I ended up using `mitmproxy`, very nice tool (https://mitmproxy.org/).

Discoveries:

System-wide CA certificate installation isn't recognized by conda's cURL. Instead it uses $PREFIX/ssl/cacert.pem as the certificate store.

Python's requests library is probably using `certifi`, which comes with another certificate store (python3.X/site-packages/certifi/cacert.pem).

Requests has a environment variable that can be set, e.g.:

```
export REQUESTS_CA_BUNDLE=/etc/ssl/certs/ca-certificates.crt
```

I had to use 

```
curl_easy_setopt(m_target, CURLOPT_CAINFO, ssl_verify.c_str());
```

instead of setting KEYFILE.

**Comment by xhochy** [comment]
@wolfv Is this also about the `proxy_servers:` option in the `.condarc`?

**Comment by wolfv** [comment]
yes, I think the proxy server options aren't correctly working with mamba (we're not configuring libcurl to mirror what conda does).
I had some prototype code that implemented these options https://github.com/mamba-org/mamba/pull/161 but not really tested it thoroughly.

**Comment by caseyclements** [comment]
Just to confirm, this is the type of error one runs into. True?

`ValueError('check_hostname needs a SSL context with either CERT_OPTIONAL or CERT_REQUIRED',)`

**Comment by wolfv** [comment]
No idea, sorry. For the proxy options you can try curl from the command line, e.g. 

curl -x http://proxy_server:proxy_port --proxy-user username:password -L http://url

**Comment by desilinguist** [comment]
It'd be great if `mamba` supported  the `proxy_servers` option in `.condarc`. We love mamba in our organization but might have to switch back to conda because we need the proxy option.

**Comment by haobibo** [comment]
Any progress on the proxy feature?

**Comment by tony-y-zhang** [comment]
my mamba installation on linux (fedora) respects proxy settings in my ~/.bashrc, YMMV

`export http_proxy='url'`
`export https_proxy='url2'`

**Comment by haobibo** [comment]
Any idea about how to setup a proxy for mamba on Windows?

**Comment by tony-y-zhang** [comment]
> Any idea about how to setup a proxy for mamba on Windows?

just tried this, manually `set`-ing my proxies and mamba was functioning in Anaconda prompt 
https://stackoverflow.com/questions/11726881/how-to-set-an-http-proxy-in-python-2-7

That said, support in condarc would be better UX

**Comment by haobibo** [comment]
Thank you @tony-y-zhang, the method you mentioned on WIndows works well.
Still, it will be better if mamba can read proxy config from config files.

> set http_proxy=http://proxy.myproxy.com
> set https_proxy=https://proxy.myproxy.com
